### ISSUE TITLE
[PATCH] set host in web.client.HTTPClientFactory

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7595]
|<img alt="thfavier's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| thfavier reported|
|-|-|
|Trac ID|trac#371|
|Type|enhancement|
|Created|2003-10-28 15:14:06Z|

Attachments:

* [httpclientfactory-tf.diff.TXT](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/355/3554dce55f341edd431fc711f6816673f081452d/32f6db2f3481312068fa21374580a94ae4226c70.TXT) (1155 bytes) - added by thfavier on 2003-10-28 15:14:06Z - 
* [twisted-1.1.1_web_client.py-tf.diff.TXT](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/355/3554dce55f341edd431fc711f6816673f081452d/decad11d3bba939fc1d2b41e6807ce9a606af7a3.TXT) (1548 bytes) - added by thfavier on 2003-12-16 17:19:46Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__371 371
type__enhancement enhancement
reporter__thfavier thfavier
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1067354046000000 1067354046000000
changetime__1469067519642369 1469067519642369
version__ 
owner__ 
cc__itamarst cc__thfavier
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="thfavier's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|thfavier commented|
|-|-|

```
#!html
<pre>
With Twisted 1.1, it is no more possible to explicitely set host in
HTTPClientFactory.

This small patch makes it possible again.

Insert classic disclaimer for first patch submission...
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="thfavier's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|thfavier commented|
|-|-|

```
#!html
<pre>
Added patch for 1.1.1
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Did my own fix - if you explicitly pass host header in headers dict, it uses that.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub commented|
|-|-|

In changeset ade1f34319a53f118e5dbb2d41e5844052bba387
```
#!CommitTicketReference repository="" revision="ade1f34319a53f118e5dbb2d41e5844052bba387"
Merge pull request #371 from scop/8641-scop-zfe-ctxmgr

Author: scop
Reviewer: rodrigc
Fixes: #8641

Implement context manager in ChunkingZipFile.readfile() zip file entries
```
